fc/mef.py

```diff
@@ -217,265 +52,152 @@ def clustering_gmm(data, n_clusters = 8, initialization = 'distance_sub',
-    `peaks_mef` that have an undetermined value (NaN), and the
-    corresponding values in peaks_ch.
-
-    Parameters
-    ----------
-    peaks_ch : array
-        Sorted fluorescence values of bead populations in channel units.
-    peaks_mef : array
-        Sorted fluorescence values of bead populations in MEF units.
-    peaks_ch_std_mult_l, peaks_ch_std_mult_r : float, optional
-        Number of standard deviations from `peaks_ch_min` and
-        `peaks_ch_max`, respectively, that a value in `peaks_ch` has to be
-        closer than to be discarded.
-    peaks_ch_min, peaks_ch_max : int, optional
-        Minimum and maximum tolerable fluorescence value in channel units.
-
-    Returns
-    -------
-    sel_peaks_ch : array
-        Selected fluorescence values of bead populations in channel units.
-    sel_peaks_mef : array
-        Selected fluorescence values of bead populations in MEF units.
-
-    """
-    # Minimum peak standard deviation will be 1.0
+    # Set minimum standard deviation to one.
     min_std = 1.0
-    peaks_ch_std = peaks_ch_std.copy()
-    peaks_ch_std[peaks_ch_std < min_std] = min_std
-
-    # Discard channel-space peaks
-    if (peaks_ch[0] - peaks_ch_std[0]*peaks_ch_std_mult_l) <= peaks_ch_min and\
-        (peaks_ch[-1] + peaks_ch_std[-1]*peaks_ch_std_mult_r) >= peaks_ch_max:
-        raise ValueError('Peaks are being cut off at both sides.')
-    elif (peaks_ch[0] - peaks_ch_std[0]*peaks_ch_std_mult_l) <= peaks_ch_min:
-        discard_ch = 'left'
-        discard_ch_n = 1
-        while (peaks_ch[discard_ch_n] - 
-            peaks_ch_std[discard_ch_n]*peaks_ch_std_mult_l) <= peaks_ch_min:
-            discard_ch_n = discard_ch_n + 1
-        sel_peaks_ch = peaks_ch[discard_ch_n:]
-    elif (peaks_ch[-1] + peaks_ch_std[-1]*peaks_ch_std_mult_r) >= peaks_ch_max:
-        discard_ch = 'right'
-        discard_ch_n = 1
-        while (peaks_ch[-1-discard_ch_n] +
-            peaks_ch_std[-1-discard_ch_n]*peaks_ch_std_mult_r) >= peaks_ch_max:
-            discard_ch_n = discard_ch_n + 1
-        sel_peaks_ch = peaks_ch[:-discard_ch_n]
-    else:
-        discard_ch = False
-        discard_ch_n = 0
-        sel_peaks_ch = peaks_ch.copy()
-
-    # Discard MEF peaks
-    discard_mef_n = len(peaks_mef) - len(sel_peaks_ch)
-    if discard_ch == 'left':
-        sel_peaks_mef = peaks_mef[discard_mef_n:]
-    elif discard_ch == 'right':
-        sel_peaks_mef = peaks_mef[:-discard_mef_n]
-    elif discard_ch == False and discard_mef_n == 0:
-        sel_peaks_mef = peaks_mef.copy()
-    else:
-        ValueError('Number of MEF values and channel peaks does not match.')
-    
-    # Discard unknown (NaN) peaks
-    unknown_mef = np.isnan(sel_peaks_mef)
-    n_unknown_mef = np.sum(unknown_mef)
-    if n_unknown_mef > 0:
-        sel_peaks_ch = sel_peaks_ch[np.invert(unknown_mef)]
-        sel_peaks_mef = sel_peaks_mef[np.invert(unknown_mef)]
+    pop_std[pop_std < min_std] = min_std
```

Comment:
I don't understand why minimum STD is set to 1.0. Is this a degenerate case? This seems to make assumptions about the data, e.g. that an STD < 1.0 is too low. In general, I'm not sure why that's justified. If you're operating on a transformed `FCSData` object in RFI space and it's a population on the low end, seems like a standard deviation of < 1.0 might be reasonable there? Actually maybe not. That starts at 10^0? = 1? So you still really shouldn't be at a scale where a STD < 1.0 is reasonable. Either way, would just be nice to state why this is done.